azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -295,23 +295,23 @@ def run(self, goal_state):
             requested_version = self.__get_requested_version(agent_family)
             agent_manifest = None  # This is to make sure fetch agent manifest once per update
             warn_msg = ""
-            if requested_version is None:
+            if requested_version is not None and agent_family.is_version_from_rsm:
```

Comment:
why is this even possible? if RSM is not requesting an update then we would simply get the version from the manifest, right?